components/region_cache_memory_engine/src/region_label.rs

```diff
@@ -0,0 +1,406 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::{sync::Arc, time::Duration};
+
+use dashmap::DashMap;
+use futures::{
+    compat::Future01CompatExt,
+    stream::{self, StreamExt},
+};
+use kvproto::meta_storagepb::EventEventType;
+use pd_client::{
+    meta_storage::{Checked, Get, MetaStorageClient, Sourced, Watch},
+    Error as PdError, PdClient, RpcClient, REGION_LABEL_PATH_PREFIX,
+};
+use serde::{Deserialize, Serialize};
+use tikv_util::{error, info, timer::GLOBAL_TIMER_HANDLE};
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, PartialEq)]
+pub struct RegionLabel {
+    pub key: String,
+    pub value: String,
+    pub ttl: Option<String>,
+    pub start_at: Option<String>,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize, PartialEq)]
+pub struct LabelRule {
```

Comment:
I see the structure in Go is 
https://github.com/tikv/pd/blob/783d060861cef37c38cbdcab9777fe95c17907fe/server/schedule/labeler/rules.go#L29C1-L49C1.
is it inconsistent?

Reply:
I'm following this: https://github.com/pingcap/tidb/blob/master/br/pkg/pdutil/pd.go#L648 (note that there's no index specified in pd.go)

Looks like I can remove ttl and start_at from LabelRule. Will do so.

Reply:
done.